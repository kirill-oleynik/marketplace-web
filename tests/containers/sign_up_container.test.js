const React = require('react');
const renderer = require('react-test-renderer');
const { SignUpContainer } = require('../../app/containers/sign_up_container');

jest.mock('../../app/components/auth/sign_up_form', () => 'Form');
jest.mock('../../app/components/auth/log_in_with_linkedin', () => 'LogInWithLinkedin');

describe('#render', () => {
  test('it renders correctly', () => {
    const tree = renderer.create(
      <SignUpContainer
        errors={{}}
        signUp={() => {}}
        t={(translation) => translation}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
