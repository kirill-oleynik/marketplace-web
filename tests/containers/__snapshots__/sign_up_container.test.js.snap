// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render it renders correctly 1`] = `
<div>
  <div
    className="mb-15"
  >
    <h2
      className="login-form__title"
    >
      title
    </h2>
    <span
      className="d-inline-block font-14 mb-10"
    >
      alreadyHaveAccount
      Â 
      <a
        className="login-form__link ml-5"
        href="/sign_in"
        onClick={[Function]}
      >
        logInHere
      </a>
    </span>
    <LogInWithLinkedin
      className="w-100 mb-20"
    />
    <div
      className="d-flex align-items-center justify-content-between mb-10"
    >
      <div
        className="login-form__divider divider divider--dark"
      />
      <span
        className="font-14 mr-20 ml-20"
      >
        or
      </span>
      <div
        className="login-form__divider divider divider--dark"
      />
    </div>
  </div>
  <Form
    errors={Object {}}
    initialState={
      Object {
        "email": "",
        "firstName": "",
        "lastName": "",
        "password": "",
        "passwordConfirmation": "",
      }
    }
    onSubmit={[Function]}
    rules={
      Object {
        "email": Array [
          Array [
            [Function],
            "common:validations.isNotEmpty",
          ],
        ],
        "firstName": Array [
          Array [
            [Function],
            "common:validations.isNotEmpty",
          ],
        ],
        "lastName": Array [
          Array [
            [Function],
            "common:validations.isNotEmpty",
          ],
        ],
        "password": Array [
          Array [
            [Function],
            "common:validations.isNotEmpty",
          ],
        ],
        "passwordConfirmation": Array [
          Array [
            [Function],
            "common:validations.isNotEmpty",
          ],
        ],
      }
    }
    validateOnChange={true}
    validateSingle={true}
  />
  <p
    className="font-14 in-black-050 mb-20"
  >
    agree
  </p>
</div>
`;
